### DISCUSSION TITLE
What's new?

### DISCUSSION THREAD

**Discussion by Bumblebee001** [GrandOrgue/GoOdf#31] [General]
Is there anywhere one can view what feature/s have been added or what bugs have been corrected when a new version is released?

**Comment by larspalo** [comment]
There's no nice release notes created for the releases (yet), but GitHub makes it fairly easy to see what has happened. Go into the release page for the latest release and use the button "Compare", to compare it with any other you want (for instance the next latest). On that page you can see every commit made since the release you compare against, and below the commit headers you even have all the changed code presented...

![ComparingReleases](https://github.com/larspalo/GOODF/assets/6173941/9696d92c-436a-40ef-824f-246917a75e07)

There are a few bug fixes, but most important is the new feature to show the currently selected panel layout, that is a graphical rendering of the panel such as it is. At the moment, don't expect it to be 100% accurate, but it's a start... Also, it's a modeless window, which means that you can make adjustments to the individual GUI Elements of that panel and see the effects of it (hopefully immediately) updated on the rendered panel version. Expect it to improve and evolve over time...

![GraphicalPanelRepresentation](https://github.com/larspalo/GOODF/assets/6173941/4253697b-7e2c-4e7d-b5bd-e3b49181e3f8)

**Comment by Bumblebee001** [comment]
Thanks. I’ll have a look at that.

You’re on the right track. I can’t wait for the “finished product” but it
is certainly a very exciting journey and a great pleasure seeing it evolve
and bloom!

Keep up the momentum.

**Comment by Bumblebee001** [comment]
I get a blank window when I click "show Panel layout"

**Reply by larspalo** [reply]
It's fixed with the v0.6.1 release. Unfortunately I forgot to change the version number for the program so on the title bar it will still read 0.6.0. But now the panel representation should work on windows too.

**Reply by larspalo** [reply]
Also something seems to be wrong with the artifact upload of the release this time. I'll look into it when I have more time.

**Reply by larspalo** [reply]
Sorry, I'll try re-running the job with a corrected version number in the sources and see if that works.

**Reply by larspalo** [reply]
Finally, the fixed release is published!

**Comment by Bumblebee001** [comment]
Still 0.6.1?

**Reply by larspalo** [reply]
Yes.

**Comment by Bumblebee001** [comment]
Observation: text on a label not correctly represented in preview (particularly font size).

**Reply by larspalo** [reply]
What exactly is wrong? I can't say I've so far seen any difference either of the font face or font size on Linux (as long as GO is not scaling things down but render at 100% size and the relevant font is available on the system). Please provide examples that I can test. But it's of course possible that wxMSW does handle fonts sligthly different...

I've fixed the possibility of having a width of an element with a smaller value than the bitmap width with the latest commit (which won't be present in last release of course). This was not working as it should earlier.

**Comment by Bumblebee001** [comment]
![image](https://github.com/larspalo/GOODF/assets/6999990/a1b2c3f5-4a25-4b83-abf0-ddc5c98ab34b)

**Reply by larspalo** [reply]
What fonts and sizes are actually used? To me it seems that the ones used by GOODF is about 2 steps larger generally in your images. But I've not seen that here so far. Could you provide me with a basic odf that exhibit the same issue (without any samples of course, just the .organ file with possibly DUMMY pipes so that it could also be loaded in GO without any extra stuff)? It might very well be that this again is something that's different in the windows version of wx than in the Linux...

**Comment by Bumblebee001** [comment]
![image](https://github.com/larspalo/GOODF/assets/6999990/fa5fe7d3-1a5c-4f50-bd83-616de30880fd)

**Comment by Bumblebee001** [comment]
![image](https://github.com/larspalo/GOODF/assets/6999990/f3a2823e-63bf-4dbd-8bf6-cbae8b745ec0)

**Reply by larspalo** [reply]
Ok, I've tested a label with similar attributes as your odf. The display on my Linux computer is the same for GOODF and GO:

![LabelComparison](https://github.com/larspalo/GOODF/assets/6173941/bcc8412c-4240-4a84-bea8-2c43809d5b7c)

Please test this odf and compare the results in GOODF and GO on your computer.

[TestLabelFontSize.zip](https://github.com/larspalo/GOODF/files/11929351/TestLabelFontSize.zip)

If you change the font scaling in windows, does GOODF and GO react differently?

**Comment by Bumblebee001** [comment]
![image](https://github.com/larspalo/GOODF/assets/6999990/20e761c4-ce10-4f3b-9537-ee2b0d0f1c10)

**Reply by larspalo** [reply]
So, obviously there's some kind of font scaling happening in windows that's different for GOODF and GO. I'll dig in the sources to try to find out what's the cause, but unfortunately it's difficult for me to test it as I don't easily have access to any reasonably modern windows computer. Running the windows executable for GOODF in wine doesn't seemt to re-create the problem.

So again: if you change the font scaling in windows, does GOODF and GO react differently?

**Comment by Bumblebee001** [comment]
I don’t understand what you mean by “changing font scaling”.

**Reply by larspalo** [reply]
Windows has a property that allows the user to change the displayed size of all fonts. Not all applications support this though or is dpi aware, and it's possible that GOODF and GO differs in this aspect. It's also possible to disable the font scaling for individual applications at will.

See for instance https://support.microsoft.com/en-gb/windows/make-text-and-apps-bigger-c3095a80-6edd-4779-9282-623c4d721d64 and https://support.microsoft.com/en-us/topic/windows-scaling-issues-for-high-dpi-devices-508483cd-7c59-0d08-12b0-960b99aa347d among different sources.

You could test right clicking on GOODF application, select Properties, select the Compatibility tab, and then select the Disable display scaling on high DPI settings check box and see if anything changes.

**Comment by Bumblebee001** [comment]
![image](https://github.com/larspalo/GOODF/assets/6999990/d755e6f2-f990-4e89-ab90-ff1742a75085)

**Comment by Bumblebee001** [comment]
![image](https://github.com/larspalo/GOODF/assets/6999990/6f5fe0d8-8547-4c11-8560-c43a86887194)

**Reply by Bumblebee001** [reply]
Application window is larger (compare to property window) and text looks much better but label font still a bit larger and bolder than expected. Why just the label? Unless there is a subtle difference, text on stops look fine and correctly displayed as in GO!

**Reply by larspalo** [reply]
In your previous images all text in GOODF was actually larger than in GO, not only the labels. The text for the elements is drawn in exactly the same way for all of them. The real headache is how I can make the windows version behave the same as it already does under Linux... I think that it must be something about the manifest and rc files used for windows and the dpi awareness.

**Comment by Bumblebee001** [comment]
Text might appear larger but I think the graphics are a lot smaller making
text look larger in comparison. Both situations might be at play.

**Reply by larspalo** [reply]
Comparing images in your post https://github.com/larspalo/GOODF/discussions/31#discussioncomment-6333010 yields that only the text is different (and strangely enough the tiling of the background bitmap seems to be slightly different). All the images for the elements are exactly the same in size. Can't say for sure with your later posts though.

**Comment by Bumblebee001** [comment]
Looking closer, even the text on the stops are smaller in GrandOrgue. You will notice that the text fits better in the small area of the stop face than is shown on GOODF.

**Reply by larspalo** [reply]
Yes, I said that above in https://github.com/larspalo/GOODF/discussions/31#discussioncomment-6343394, but since it's only happening in the windows version there's some font scaling in the OS itself that differ in how it affects GOODF and GO. Under Linux, the fonts are exactly the same in GOODF and GO.

**Reply by Bumblebee001** [reply]
If it helps these are my system settings (which I expect to apply equally to everything):
![image](https://github.com/larspalo/GOODF/assets/6999990/5a66366e-a82e-4bf1-acb8-b6d358fdff87)

**Reply by larspalo** [reply]
> (which I expect to apply equally to everything)

And which it's most likely not doing. If you have the test odf open both in GOODF and GO and play with the scale factor, what happens? Also, you should have more detailed scale options separately for text and other elements if you dig a little deeper in the settings.

**Reply by Bumblebee001** [reply]
Altering text size from 120% to 140% did not change anything that I could notice on my sample-set. 
Both software show same text size with GOODF being slightly larger, as before.
I'll try your test file next.

**Comment by Bumblebee001** [comment]
Text set at 120%:
![image](https://github.com/larspalo/GOODF/assets/6999990/37664176-d371-4740-9827-9c8ef515d34e)

**Comment by Bumblebee001** [comment]
set at 160%: 
![image](https://github.com/larspalo/GOODF/assets/6999990/cbba319c-9e75-4df6-933c-c2ecca67a4df)

**Reply by Bumblebee001** [reply]
Desktop looks ugly with text of my folders and program icons far too big!

**Reply by Bumblebee001** [reply]
I re-set text size as before (133%).

**Comment by Bumblebee001** [comment]
I tried your test software:
Text in label look OK. Stop text slightly larger.
![image](https://github.com/larspalo/GOODF/assets/6999990/c8f923c7-50f7-4049-be10-676a0663f416)

**Reply by Bumblebee001** [reply]
It seems like you corrected the issue for the label.

**Reply by larspalo** [reply]
Ok, thanks for reporting! I'll apply the fix for all types of text on the representation panel then.

**Reply by Bumblebee001** [reply]
It is my pleasure to assist in the development of this software. I can bet you I am more excited than you are! ;-) You are fulfilling a dream I had years ago but I don't know how to program; a friend of mine who does, had dashed all my hopes.

So feel free to ask for any help I can offer.

**Reply by larspalo** [reply]
Version 0.6.2 is building right now. It has been many (really, really many) hours of work to get this software to its current state, but I think I've learned quite a lot myself too in this process! Also I think that it has been a benefit for GrandOrgue as project that I've needed to research a lot of its code too to understand more about how it's working under the hood... But the main benefit, I hope, will be for sample set producers - to have a tool that will ease the (fairly large) workload when creating sample sets for GrandOrgue - which also will benefit all users in the end.
